src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,61 @@
 package core.basesyntax;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String supplyKey = "supply";
+    private static final String buyKey = "buy";
+    private static final String inputDelimiter = ",";
+    private static final String outputDelimiter = ",";
+    private static final String newline = System.lineSeparator();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> lines;
+
+        try {
+            lines = Files.readAllLines(Path.of(fromFileName));
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot read from file " + fromFileName);
+        }
+
```

When catching `IOException` from reading the file you rethrow `new RuntimeException("Cannot read from file " + fromFileName);` but do not include the original exception as the cause. The checklist explicitly advises not to ignore exceptions and to rethrow with the original exception as the cause. Quote from checklist: "Don't ignore exceptions\nLeaving an empty catch block or `e.printStackTrace` here is a bad practice. \nBetter re-throw `RuntimeException` with the original exception and some message in the parameters:\n```\ncatch (Exception e) {\n    throw new RuntimeException("Can't read data from the file " + fileName, e);\n}\n```" Please change the catch to include the caught exception (e) as the cause, e.g. `throw new RuntimeException("Cannot read from file " + fromFileName, e);`.